cdap-api/src/main/java/co/cask/cdap/api/data/format/StructuredRecord.java

```diff
@@ -56,6 +56,10 @@ public Schema getSchema() {
     return (T) fields.get(fieldName);
   }
 
+  public Map<String, Object> getFields() {
+    return fields;
```

Comment:
making it immutable also means we can't add a set method for a field, which we may want to add for transforms that work on a single field.  I think it's better just to use get() and not expose a way to get all fields.